src/sidebar/app/utils/utils.js

```diff
@@ -11,4 +11,20 @@ function formatFooterTime(date) {
   });
 }
 
-export { formatFooterTime };
+/**
+ * 
+ * @param {HTMLElement Object} parentElement 
+ * @returns {HTMLElement Object}
+ */
+function getFirstNonEmptyElement(parentElement) {
+  // create an Array from parentElement's `children` (HTMLCollection)
+  const parentElementChildrenArray = Array.prototype.slice.call(parentElement.children);
+  // search for first child element that is not empty and return it
+  for (const [index, childElement] of parentElementChildrenArray.entries()) {
```

Comment:
limit this to like 15 - 25 parentElement children, so avoid issues with big notepads, if we cannot extract title from that then we can just use blank.html